pkg/terraform/run.go

```diff
@@ -17,10 +17,20 @@ package terraform
 import (
 	"context"
 	"io"
+	"slices"
 
+	"github.com/abcxyz/guardian/internal/version"
 	"github.com/abcxyz/guardian/pkg/child"
 )
 
+// overrideEnvVars are the environment variables to inject into the Terraform
+// child process, no matter what the user configured. These take precedence
+// over all other configurables.
+var defaultOverrideEnvVars = []string{
+	"GOOGLE_TERRAFORM_USERAGENT_EXTENSION=" + version.UserAgent,
```

Comment:
I know this is out of scope for this PR. Perhaps someday, a cleaner API for environment variables wouldn't rely on parsing the equals sign (as happens in the `environ` function), but would pass keys and values separately.

It could be a map, like:
```
myEnvVars := map[string]string{
  "MY_ENV_VAR_1": "some_value",
```

Or a list of pairs, like:
```
myEnvVars := [][]string{{"ENV_VAR_1", "value1"}, {"ENV_VAR_2", "value2"}
```

When I saw these equals signs I was confused at what could possibly be happening, until I discovered that we implemented our own little parser for environment vars using `strings.SplitN` which seems a bit much.

Reply:
Could be cleaner, just followed existing impl. We can discuss offline.